apps/functions/ng-dev/ng-dev-token.ts

```diff
@@ -0,0 +1,57 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import {CallableContext} from 'firebase-functions/lib/providers/https';
+
+/**
+ * Request a short lived ng-dev token.  If granted, we rely on session cookies as this token.  The token
+ * is to be used for all requests to the ng-dev service.
+ */
+export const ngDevTokenRequest = functions.https.onCall(
+  async ({idToken}: {idToken: string}, context: CallableContext) => {
+    if (!context.auth) {
+      // Throwing an HttpsError so that the client gets the error details.
+      throw new functions.https.HttpsError(
+        'unauthenticated',
+        'Requesting an ng-dev token requires authentication',
+      );
+    }
+    const {auth_time} = await admin.auth().verifyIdToken(idToken, /* checkRevoked */ true);
+
+    // Only allow creation if the user signed in in the last minute. We use one minute, as this
+    // token should be immediately requested upon login, rather than using a long lived session.
+    if (new Date().getTime() / 1000 - auth_time > 1 * 60) {
+      throw new functions.https.HttpsError(
+        'permission-denied',
+        'ng-dev tokens must be requested within one minute of verifying login',
+      );
+    }
+
+    /** Twenty hours in ms. */
+    const twentyHours = 1000 * 60 * 60 * 20;
```

Comment:
Optional nit: Want to move that into a constant. or inline it? doesn't seem like it needs to be stored here in a local variable.